Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -2997,25 +3163,31 @@ RefPtr<Identifier const> Parser::parse_lexical_binding()
 
     if (match_identifier()) {
         auto name = consume_identifier().DeprecatedFlyString_value();
-        return create_ast_node<Identifier>(
+        auto id = create_ast_node<Identifier>(
             { m_source_code, binding_start.position(), position() },
             name);
+        m_state.current_scope_pusher->register_identifier(const_cast<Identifier&>(*id));
```

Comment:
Feels like we should have a helper that does both `create_ast_node<Identifier>` and `register_identifier`, so you don't have to remember to do it everywhere.